src/resources/cubemap.js

```diff
@@ -297,7 +297,7 @@ Object.assign(CubemapHandler.prototype, {
             } else if (self.compareAssetIds(assetId, loadedAssetIds[i])) {
                 // asset id hasn't changed from what is currently set
                 onReady(i, loadedAssets[i]);
-            } else if (parseInt(assetId, 10) === assetId) {
+            } else if (parseInt(assetId, 10) == assetId) {
```

Comment:
This is a bit scary. I really dislike double equals and so does ESlint it seems:

https://eslint.org/docs/rules/eqeqeq

Some coder in the future might just decide to 'improve' the code and switch to triple equals, not understanding the rationale. Is there not a clearer way to deal with this?

Reply:
I can add a comment with explanation?

Reply:
something like:
```javascript
} else if (parseInt(assetId, 10) == assetId) {  // NOTE: use '==' to handle assetId being 123 or '123'.
```

Reply:
wrapped the test in a function and used explicit '===' instead